quant docs: clean up hardswish

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40377 quant docs: add and clean up ELU
* #40346 quant docs: add and clean up BatchNorm{n}d
* #40345 quant docs: add and clean up InstanceNorm{n}d
* #40343 quant docs: add and clean up GroupNorm
* #40342 quant docs: add and clean up LayerNorm
* #40341 quant docs: add and clean up hardtanh
* #40340 quant docs: add and clean up hardsigmoid
* **#40323 quant docs: clean up hardswish**

Cleans up the naming and the function param docs for quantized hardswish.
Remove redundant docstrings and link to floating point modules instead.

Test plan:
* build the docs on Mac OS
* verify that every link works as expected

Differential Revision: [D22152638](https://our.internmc.facebook.com/intern/diff/D22152638)

vkuzo committed:
quant docs: clean up hardswish

Cleans up the naming and the function param docs for quantized hardswish.
Remove redundant docstrings and link to floating point modules instead.

Test plan:
* build the docs on Mac OS
* verify that every link works as expected

[ghstack-poisoned]

jerryzh168 commented on torch/nn/quantized/functional.py:
```diff
@@ -427,26 +427,12 @@ def hardtanh(input, min_val=-1., max_val=1., inplace=False):
-            x & \text{if~} x \ge +3, \\
-            x^2/6 & \text{otherwise}
-        \end{cases}
+    r"""This is the quantized version of :meth:`torch.nn.functional.hardswish`.
```
probably should be :meth:`~torch.nn.functional.hardswish`

vkuzo replied:
it works and is used in pytorch codebase both ways, but sure

jerryzh168 commented on docs/source/quantization.rst:
```diff
@@ -193,7 +193,7 @@ Basic activations are supported.
 * :meth:`~torch.nn.functional.avg_pool2d` - Average pooling
 * :meth:`~torch.nn.functional.interpolate` - Interpolation
-* :meth:`~torch.nn.functional.hardswish` - Hard Swish
+* :meth:`~torch.nn.functional.hardswish` - Hardswish
```
actually why are these method? I feel it makes more sense for these to be `func`

vkuzo replied:
not sure if there is any difference in the generated docs.  If this is important, might be good to be a separate PR on the entire file, so that it's consistent within the file

jerryzh168 replied:
yeah, it should be changed in a separate PR I think

vkuzo committed:
Update on "quant docs: clean up hardswish"


Cleans up the naming and the function param docs for quantized hardswish.
Remove redundant docstrings and link to floating point modules instead.

Test plan:
* build the docs on Mac OS
* verify that every link works as expected

[ghstack-poisoned]

vkuzo committed:
Update on "quant docs: clean up hardswish"


Cleans up the naming and the function param docs for quantized hardswish.
Remove redundant docstrings and link to floating point modules instead.

Test plan:
* build the docs on Mac OS
* verify that every link works as expected

[ghstack-poisoned]

vkuzo committed:
Update on "quant docs: clean up hardswish"


Cleans up the naming and the function param docs for quantized hardswish.
Remove redundant docstrings and link to floating point modules instead.

Test plan:
* build the docs on Mac OS
* verify that every link works as expected

Differential Revision: [D22152638](https://our.internmc.facebook.com/intern/diff/D22152638)

[ghstack-poisoned]

vkuzo committed:
Update on "quant docs: clean up hardswish"


Cleans up the naming and the function param docs for quantized hardswish.
Remove redundant docstrings and link to floating point modules instead.

Test plan:
* build the docs on Mac OS
* verify that every link works as expected

Differential Revision: [D22152638](https://our.internmc.facebook.com/intern/diff/D22152638)

[ghstack-poisoned]

vkuzo committed:
Update on "quant docs: clean up hardswish"


Cleans up the naming and the function param docs for quantized hardswish.
Remove redundant docstrings and link to floating point modules instead.

Test plan:
* build the docs on Mac OS
* verify that every link works as expected

Differential Revision: [D22152638](https://our.internmc.facebook.com/intern/diff/D22152638)

[ghstack-poisoned]

vkuzo committed:
Update on "quant docs: clean up hardswish"


Cleans up the naming and the function param docs for quantized hardswish.
Remove redundant docstrings and link to floating point modules instead.

Test plan:
* build the docs on Mac OS
* verify that every link works as expected

Differential Revision: [D22152638](https://our.internmc.facebook.com/intern/diff/D22152638)

[ghstack-poisoned]
